lib/mozart_fetcher/timeout_parser.ex

```diff
@@ -6,11 +6,13 @@ defmodule MozartFetcher.TimeoutParser do
     query(uri.query)
   end
 
-  def max_timeout(components) do
+  def max(components) do
     components
-    |> Enum.map(&Map.get(&1, :endpoint))
-    |> Enum.map(fn endpoint -> parse(endpoint) end)
-    |> Enum.max()
+    |> Enum.reduce(@default_timeout, &component_timeout_or_current_timeout/2)
```

Comment:
why this?

Reply:
so we assume that there's a performance gain with no verification.

Reply:
It should be more performant. Isn't this part more of an opinion on the style.

Reply:
ok

Reply:
let's merge it

Reply:
we assumed it wasn't a performance hit before, with no verification